config/config.go

```diff
@@ -75,10 +141,100 @@ type API struct {
 	FreeSpaceRequiredBytes int64  `toml:"free_space_required_bytes"`
 }
 
-type Transport struct {
-	Enabled                bool   `toml:"enabled"`
-	Workers                int    `toml:"workers"`
-	SleepSeconds           int    `toml:"sleep_seconds"`
-	DataDir                string `toml:"data"`
-	FreeSpaceRequiredBytes int64  `toml:"free_space_required_bytes"`
+// decodeTransportMemory decodes the [transport] config for `[transport.type]` = `memory`
+func decodeTransportMemory(metaData toml.MetaData, confData configData) (transport.DataTransport, error) {
```

Comment:
This logic should live in the `main` package. Let's keep the `config` package just focused on marshaling or validating configuration files themselves.

This is what I would propose (in pseudocode):

``` go
var queue Queue
if config.SQS {
  queue = NewSQSQueue(config.SQS)
} else if config.Redis {
  queue = NewRedisQueue(config.Redis)
} else {
  queue = NewDummyQueue() // Dummy implementation which does nothing or returns an error
}

var storage Storage 
if config.S3 {
  storage = NewS3(...)
} // etc

var transport Transport
switch (config.Transport.Type) {
  case: "queuestorage":
     transport = NewQueueStorageTransport(queue, storage)
  default:
    // dummy transport which returns an error
}
```

Finally, rather than a `Memory` transport, let's instead have both a Queue and Storage type of Memory, and reuse the QueueStorage transport.